src/server/generic_family.cc

```diff
@@ -463,6 +458,209 @@ void GenericFamily::Stick(CmdArgList args, ConnectionContext* cntx) {
+    }
+    if constexpr (std::is_assignable_v<KEY_TYPE&, std::string&&>) {
+      key = std::move(item);
+    } else {
+      VLOG(0) << "Got string when expected int";
+      assert(false && "unreachable");
+    }
+    return true;
+  }
+
+  bool Parse(int64_t item) {
+    if constexpr (STORE_SCORE) {
+      this->score = item;
+    }
+    if constexpr (std::is_assignable_v<KEY_TYPE&, int64_t>) {
+      key = item;
+    } else {
+      VLOG(0) << "Got string when expected string";
+      assert(false && "unreachable");
+    }
+    return true;
+  }
+
+  COMPARE_TYPE Cmp() const {
+    if constexpr (STORE_SCORE) {
+      return this->score;
+    } else {
+      return key;
+    }
+  }
+
+  template <typename F> void visit(F&& f) const {
+    // TODO: better check
+    if constexpr (!ALPHA && STORAGE_KIND == container_utils::StorageKind::MIXED)
+      std::visit(std::forward<F>(f), key);
+    else
+      f(key);
+  }
+};
+
+namespace helper {
+// Sort entry indices are encoded with the formula ALPHA * 3 + STORAGE
+template <size_t I, bool ALPHA = I / 3, size_t STORAGE = I % 3> static auto MapType() {
+  return SortEntry<ALPHA, container_utils::StorageKind{STORAGE}>{};
+}
+
+template <size_t... seq> static auto ListTypes(std::index_sequence<seq...>) {
+  return std::variant<std::vector<decltype(MapType<seq>())>...>{};
+}
+}  // namespace helper
+
+// std::variant of all possible vector<SortEntry<*, *>>
+using SortEntryList = decltype(helper::ListTypes(std::make_index_sequence<6>{}));
+
+namespace helper {
+template <int I> static SortEntryList MakeInternal(size_t i) {
+  if (I == i) {
+    return SortEntryList{std::in_place_index<I>};
+  } else if constexpr (I + 1 < 6) {
+    return MakeInternal<I + 1>(i);
+  } else {
+    return SortEntryList{};
+  }
+}
+}  // namespace helper
+
+// Create empty SortEntryList with type deduced from runtime arguments
+static SortEntryList MakeSortEntryList(bool alpha, container_utils::StorageKind storage) {
+  // See helper::MapType for the formula
+  return helper::MakeInternal<0>(size_t(alpha) * 3 + size_t(storage));
+}
+
+OpResult<SortEntryList> OpFetchSortEntries(const OpArgs& op_args, std::string_view key,
+                                           bool alpha) {
+  using namespace container_utils;
+
+  auto [it, _] = op_args.shard->db_slice().FindExt(op_args.db_ind, key);
+  if (!IsValid(it) || !IsContainer(it->second)) {
+    return OpStatus::KEY_NOTFOUND;
+  }
+
+  auto storage_kind = GetStorageKind(it->second);
+  auto result = MakeSortEntryList(alpha, storage_kind);
+  bool success = std::visit(
+      [&pv = it->second](auto& entries) {
+        entries.reserve(pv.Size());
+        return Iterate(pv, [&entries](auto&& val) {
+          return entries.emplace_back().Parse(std::forward<decltype(val)>(val));
+        });
+      },
+      result);
+  return success ? OpResult{std::move(result)} : OpStatus::WRONG_TYPE;
+}
+
+void GenericFamily::Sort(CmdArgList args, ConnectionContext* cntx) {
```

Comment:
This function is almost 80 lines long, maybe it'll be more readable when splitted